dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/C3Renderer.java

```diff
@@ -169,11 +176,11 @@ private C3Displayer getAreaChartForSubType(DisplayerSubType subtype) {
 
     @Override
     public List<DisplayerType> getSupportedTypes() {
-        return Arrays.asList(LINECHART, BARCHART, PIECHART, AREACHART, BUBBLECHART);
+        return SUPPORTED_TYPES;
     }
     
     @Override
     public boolean isDefault(DisplayerType type) {
-        return false;
+        return SUPPORTED_TYPES.contains(type);
```

Comment:
Thanks for you precise analysis again, @jhrcek I really appreciate.

The issue testing cross renderers is that they are decoupled and registered using CDI in RendererManager ( https://github.com/kiegroup/appformer/blob/master/dashbuilder/dashbuilder-client/dashbuilder-displayer-client/src/main/java/org/dashbuilder/displayer/client/RendererManager.java#L74 ).

if two renderers wants to be default for a displayerType, then there's no way to know which one will be used, except if we force a default using the system property `org.dashbuilder.renderer.default`. 

Please let me know if you think that it is still possible to test it.

Thanks.

Reply:
Ok, I reviewe the RendererManager code that you linked and it seems that the wors thing that can happen in case that 2 renderers are (accidentally) declared as default for given displayer type, is that we might get some non-determinism. If the set of renderers available in the workbench isn't expected to change too much, I'd still recommend creating a test for "the coverage of all types by all renderers". If you think there would be no added value, you can leave it as it is now.